lib/ticks-to-tree.js

```diff
@@ -99,71 +112,184 @@ function ticksToTree (ticks, options = {}) {
           type = 'WASM'
         }
       }
-
       if (type && type !== 'JS') name += ' [' + type + (kind ? ':' + kind : '') + ']'
-      return { S, name, value: 0, top: 0 }
+      return { S, name, type, value: 0, top: 0 }
     })
-
     labelInitFrames(stack, regExp)
 
-    addToMergedTree(stack.map(({ S, name, value, top }) => ({ S, name, value, top })))
+    addToMergedTree(trees, stack.map(({ S, name, type, value, top }) => ({ S, name, type, value, top })))
     // mutate original (save another loop over stack + extra objects)
-    addToUnmergedTree(stack)
+    addToUnmergedTree(trees, stack)
   })
 
-  function addToMergedTree (stack) {
-    var lastFrame = null
-    stack.forEach((frame, ix) => {
-      // TODO need an elegant way to place inlined frames into *other* stacks
-      // where the function has disappeared – for now we just remove inlined
-      // stacks from merged graphs
-      if (frame.S > 2) return // skip inlined
-      if (ix > 0) lastFrame.children = lastFrame.children || []
-      const children = (ix === 0) ? merged.children : lastFrame.children
-      const child = children.find(({ name }) => name === frame.name)
-
-      if (child === undefined) children.push(frame)
-      else frame = child
-
-      if (ix === stack.length - 1) frame.top++
-      if (ix === 0) merged.value += 1
-      frame.value++
-
-      lastFrame = frame
+  if (sourceMaps) {
+    await Promise.all(Object.keys(jsLibraries).map(async (jsLibrary) => {
+      return loadSourceMap(jsLibrary, sourceMaps)
+    }))
+  }
+  if (sourceMaps || relativePath) {
+    Object.keys(trees).map((tree) => {
+      trees[tree].children.map((frame, index) => {
+        renameFrame(options, frame)
+      })
     })
   }
+  Object.keys(loadedSourceMaps).forEach((file) => {
+    if (loadedSourceMaps[file] !== false) {
+      loadedSourceMaps[file].destroy()
+    }
+  })
+
+  return trees
+}
+
+const loadedSourceMaps = {}
```

Comment:
This should not be a global. It creates a memory leak

Reply:
No problem. I'll change this.

Reply:
@mcollina Do you have any suggestions for this? loadedSourceMaps is used in 3 different functions.

Reply:
You need to pass it as a parameter.